Nodejs/Product/Npm/ReaderPackageJsonSource.cs

```diff
@@ -17,12 +17,21 @@
 using System;
 using System.IO;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.NodejsTools.Npm {
     public class ReaderPackageJsonSource : IPackageJsonSource {
         public ReaderPackageJsonSource(TextReader reader) {
             try {
-                Package = JsonConvert.DeserializeObject(reader.ReadToEnd());
+                var text = reader.ReadToEnd();
+                try {
+                    // JsonConvert and JObject.Parse exhibit slightly different behavior,
+                    // so fall back to JObject.Parse if JsonConvert does not properly deserialize
+                    // the object.
+                    Package = JsonConvert.DeserializeObject(text);
+                } catch (ArgumentException) {
+                    Package = JObject.Parse(text);
+                }

```

Comment:
Yes, it is safe because DeserializeObject actually returns JObject under the covers when no explicit type is specified (`JsonConvert.DeserializeObject(...).GetType().FullName` returns "Newtonsoft.Json.Linq.JObject")

It's unclear why the two follow different codepaths, though. Perhaps they've fixed this issue in later versions of Newtonsoft.Json, but this workaround will have to suffice for now.